rfcs/20220221-3d_model_support.md

```diff
@@ -0,0 +1,181 @@
+
+- When a unique identifier is necessary, we should use the string constant `const MODEL_3D = '3d model'` to refer to 3D models. Compare this to `AUDIO` for audio files, for example.
+- Similar to the catalog/API, 3D model support should be enabled via a feature flag. This work should be able to be committed directly to `main` without a base feature branch needed. Meta search support for 3D models does not need to be feature flagged.
+- e2e tests should be added for any new routes; and updated for existing routes that are modified. PRs will be blocked in the abcense of these tests.
+- New content types should always launch in `beta` until we reach a certain number of results and/or are confident in our UI components (this metric(s) need to be determined).
+- Many of the todos identified here can be broken down into even smaller pieces in the future, particularly around the user interface.
+
+## Technical implementation
+
+### 1. Add 3D Models to Meta Search
+
+#### Prerequisites
+
+- Need a list of 3D model providers to use in meta search. This could be a mix of sources we intend to use in the Catalog but also some that don't have APIs and will stay in meta search after launching 3D model support.
+
+#### Todos
+
+- [ ] Add 3D models to the content switcher components:
+  - [ ] https://github.com/WordPress/openverse-frontend/blob/bbccc36ed49186b369cc4a69693fbc6d1411f29a/src/components/VContentSwitcher/VContentSwitcherButton.vue
+  - [ ] https://github.com/WordPress/openverse-frontend/blob/7d74f033ddc4d35889d7c43756c36150beee4192/src/components/VContentSwitcher/VContentTypes.vue
+- [ ] Create a `model/index.vue` endpoint for 3D models in the [pages directory](https://github.com/WordPress/openverse-frontend/blob/037f2efb768fdbdcff4b875e7f7c271ff50df59f/src/pages)
+- [ ] Add 3D model sources to the [legacy sources lookup](https://github.com/WordPress/openverse-frontend/blob/268532a05004b72fb91b77a875f3a3c1d7fb0599/src/utils/get-legacy-source-url.js#L1)
+- [ ] Add 3D models to the [media constants](https://github.com/WordPress/openverse-frontend/blob/4aeb9fd0d9f893f2df1752d6bb1e52351b8c3d35/src/constants/media.js#L13) file, making sure it is set to the `ADDITIONAL` status.
+
+### 2. Setup the data for 3D models
+
+- [ ] Create a feature flag for 3D model support. All 3D model features will be behind this.
+- [ ] Generate mock/sample api response data for 3D models.
+- [ ] Create a new media service for Models in the [data directory](https://github.com/WordPress/openverse-frontend/blob/fe24ec9bf6a4d3ad429901cef53744dccd93ed5d/src/data)
+- [ ] Update the [media constants](https://github.com/WordPress/openverse-frontend/blob/4aeb9fd0d9f893f2df1752d6bb1e52351b8c3d35/src/constants/media.js#L1) file with new constants for models, and under the `BETA` flag.
+- [ ] Update the [media store](https://github.com/WordPress/openverse-frontend/blob/f9f549beace4f76480c8cbf1e91d8940a0aeef2b/src/store/media.js#L347) with 3D model support.
```

Comment:
This link will probably stale depending on the timing with #174's implementation.